cni/pkg/ambient/informers.go

```diff
@@ -103,20 +116,20 @@ func (s *Server) Reconcile(input any) error {
 		wasEnabled := oldPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionEnabled
 		nowEnabled := ambientpod.PodZtunnelEnabled(ns, newPod)
 		if wasEnabled && !nowEnabled {
-			log.Debugf("Pod %s no longer matches, removing from mesh", newPod.Name)
+			log.Debugf("Pod no longer matches, removing from mesh")
 			s.DelPodFromMesh(newPod)
 		}
 
 		if !wasEnabled && nowEnabled {
-			log.Debugf("Pod %s now matches, adding to mesh", newPod.Name)
+			log.Debugf("Pod now matches, adding to mesh")
```

Comment:
In previous experimental branch, we used "oldPod == newPod" in update Event to identify if it's intended to force update the pod. 

Will current change resolve ztunnel restart case? Since for iptables, we cleared rules while ztunnel restart,  should no Add event for this case,  and update Event will do nothing for this case.

Reply:
If we clear rules on ztunnel update then we need to make sure we re-enqueue all pods on the node when ztunnel comes up, unconditionally.

In my test this does appear to be broken..

Reply:
This all worked ~1 month ago. We should add tests.

Reply:
As mentioned, `EventUpdate` _as well as_ `EventAdd` need the `nowEnabled && !IsPodInIpset` check.

Without that addition, this PR doesn't fix https://github.com/istio/ztunnel/issues/521, with that fix, it does.